### PR TITLE
Introduce ESAI params support

### PR DESCRIPTION
**Pull request by dbaluta**
Parameters for ESAI are read from topology file by the Linux driver and sent to FW on DSP.

This is Linux kernel counterpart of PR: https://github.com/thesofproject/sof/pull/1892

### CODE REVIEW

**Comment by ranj063** [comment]
@dbaluta looks good. The file dai-imx.h is missing Author information though.

**Comment by dbaluta** [comment]
@ranj063 Fixed now.

File: include/sound/sof/dai-imx.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/* SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause) */
+
+/*
+ * Copyright 2019 NXP
+ *
+ * Author: Daniel Baluta <daniel.baluta@nxp.com>
+ *
```
Comment (paulstelian97) [inline include/sound/sof/dai-imx.h]: Is this extra line necessary?

Comment (dbaluta) [reply include/sound/sof/dai-imx.h]: You are right. Not needed. Fixed now.

**Comment by dbaluta** [comment]
@plbossart, @paulstelian Yes, will update the ABI just doing more testing please lets postpone merging the 2 PRs for now

**Comment by dbaluta** [comment]
> aren't we missing the ABI change from 10 to 11?

Fixed now.

**Comment by dbaluta** [comment]
@plbossart @paulstelian97 everything is ready now!

**Pull request merged**